src/urllib3/connection.py

```diff
@@ -493,6 +494,56 @@ def getresponse(  # type: ignore[override]
         return response
 
 
+class _HTTP2ProbeCache:
```

Comment:
It might make more sense to move this whole object to the `http2` module, since it's not useful at all for non-HTTP/2 cases.

Reply:
I'm not super happy with what I've done to remove the circular dependency, let me know if you have better ideas.

Reply:
What part feels wrong? I'm happy with the current solution. The probe cache is self-contained. (The way we expose methods as module functions is slightly surprising, but is still readable in http2/connection.py)